clients/redis/redis.go

```diff
@@ -0,0 +1,298 @@
+package redis
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+
+	"github.com/artie-labs/transfer/lib/config"
+	"github.com/artie-labs/transfer/lib/db"
+	"github.com/artie-labs/transfer/lib/destination"
+	"github.com/artie-labs/transfer/lib/destination/types"
+	"github.com/artie-labs/transfer/lib/kafkalib"
+	"github.com/artie-labs/transfer/lib/optimization"
+	sqllib "github.com/artie-labs/transfer/lib/sql"
+)
+
+const (
+	artieEmittedAtField = "_artie_emitted_at"
+	artieDataField      = "_artie_data"
+	streamMaxLen        = 1000 // Auto trim to 1000 entries
+)
+
+var retryableNetworkErrors = []error{
+	syscall.ECONNRESET,
+	syscall.ECONNREFUSED,
+	io.EOF,
+	syscall.ETIMEDOUT,
+}
+
+// isRetryableNetworkError checks for common network errors that are retryable
+func isRetryableNetworkError(err error) bool {
```

Comment:
This already exists in https://github.com/artie-labs/transfer/blob/5f3d8148e437d6446e1a08d7aa158bc4db4e285a/lib/db/errors.go#L11, can we reuse?

Reply:
`lib/db` maybe doesn't make sense and we can move it out?

Reply:
Do it!